dynawo/sources/API/PAR/PARParametersSetCollection.cpp

```diff
@@ -47,11 +48,45 @@ ParametersSetCollection::addParametersSet(shared_ptr<ParametersSet> paramSet, bo
   parametersSets_[id] = paramSet;
 }
 
+void
+ParametersSetCollection::addMacroParameterSet(shared_ptr<MacroParameterSet> macroParamSet, bool force) {
+  assert(macroParamSet && "impossible to add null macro parameter set pointer to collection");
+  string id = macroParamSet->getId();
+  if (hasMacroParametersSet(id)) {
+    if (force) {
+      // @todo : force / Create custom id, change parameters set one and adds it to the map
+    } else  {
+      throw DYNError(Error::API, MacroParameterSetAlreadyExists, id);
+    }
+  }
+  macroParametersSets_[id] = macroParamSet;
+}
+
 shared_ptr<ParametersSet>
 ParametersSetCollection::getParametersSet(const string& id) {
   map< string, shared_ptr<ParametersSet> >::iterator itParamSet = parametersSets_.find(id);
   if (itParamSet == parametersSets_.end())
     throw DYNError(DYN::Error::API, ParametersSetNotFound, id);
+  // if a macroParSet is defined in the set, we add the references and parameters associated
+  if (itParamSet->second->hasMacroParSet()) {
+    map< string, shared_ptr<MacroParSet> >::iterator itMacroParSet;
+    for (itMacroParSet = itParamSet->second->getMacroParSets().begin(); itMacroParSet != itParamSet->second->getMacroParSets().end(); itMacroParSet++) {
```

Comment:
I generally write my for loops as follows for performance reasons:
for (map< string, shared_ptr<MacroParSet> >::iterator itMacroParSet = itParamSet->second->getMacroParSets().begin(), **itEnd = itParamSet->second->getMacroParSets().end()**; itMacroParSet != **itEnd**; **++i**tMacroParSet)